core/src/avm1/globals/bitmap_data.rs

```diff
@@ -800,13 +800,90 @@ pub fn copy_pixels<'gc>(
 }
 
 pub fn merge<'gc>(
-    _activation: &mut Activation<'_, 'gc, '_>,
+    activation: &mut Activation<'_, 'gc, '_>,
     this: Object<'gc>,
-    _args: &[Value<'gc>],
+    args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error<'gc>> {
     if let Some(bitmap_data) = this.as_bitmap_data_object() {
         if !bitmap_data.disposed() {
-            log::warn!("BitmapData.merge - not yet implemented");
+            let source_bitmap = args
+                .get(0)
+                .unwrap_or(&Value::Undefined)
+                .coerce_to_object(activation);
+
+            let source_rect = args
+                .get(1)
+                .unwrap_or(&Value::Undefined)
+                .coerce_to_object(activation);
+
+            let src_min_x = source_rect
+                .get("x", activation)?
+                .coerce_to_i32(activation)?;
+            let src_min_y = source_rect
+                .get("y", activation)?
+                .coerce_to_i32(activation)?;
+            let src_width = source_rect
+                .get("width", activation)?
+                .coerce_to_i32(activation)?;
+            let src_height = source_rect
+                .get("height", activation)?
+                .coerce_to_i32(activation)?;
+
+            let dest_point = args
+                .get(2)
+                .unwrap_or(&Value::Undefined)
+                .coerce_to_object(activation);
+
+            let dest_x = dest_point.get("x", activation)?.coerce_to_i32(activation)?;
+            let dest_y = dest_point.get("y", activation)?.coerce_to_i32(activation)?;
+
+            let red_mult = args
+                .get(3)
+                .unwrap_or(&Value::Undefined)
+                .coerce_to_u16(activation)?;
+
+            let green_mult = args
+                .get(4)
+                .unwrap_or(&Value::Undefined)
+                .coerce_to_u16(activation)?;
+
+            let blue_mult = args
+                .get(5)
+                .unwrap_or(&Value::Undefined)
+                .coerce_to_u16(activation)?;
+
+            let alpha_mult = args
+                .get(6)
+                .unwrap_or(&Value::Undefined)
+                .coerce_to_u16(activation)?;
```

Comment:
Similarly, I think this should be `coerce_to_i32(activation)? as u16`, cause it seems like these wrap to negativate at 2^31:
```
// => redMult clamps to 256
b2.merge(b, new Rectangle(0,0,200,200), new Point(0, 0), 2147483648, 0.0, 0.0, 0.0);
// => redMult clamps to 0
b2.merge(b, new Rectangle(0,0,200,200), new Point(0, 0), 2147483648, 0.0, 0.0, 0.0);
```